enterprise/micronaut/test/unit/src/org/netbeans/modules/micronaut/gradle/MicronautGradleArtifactsImplTest.java

```diff
@@ -40,6 +44,14 @@
  */
 public class MicronautGradleArtifactsImplTest extends NbTestCase {
 
+    static {
+        // TODO remove ASAP from MicronautGradleArtifactsImplTest and ProjectViewTest
+        // investigate "javax.net.ssl.SSLHandshakeException: Received fatal alert: handshake_failure"
+        // during gradle download "at org.netbeans.modules.gradle.spi.newproject.TemplateOperation$InitStep.execute(TemplateOperation.java:317)"
+        // this looks like a misconfigured webserver to me
+        System.setProperty("https.protocols", "TLSv1.2");
+    }
```

Comment:
webserver does not support TLS 1.3:
https://www.ssllabs.com/ssltest/analyze.html?d=services.gradle.org&s=2606%3a4700%3a0%3a0%3a0%3a0%3a6811%3a8125

but it appears as if the download tries it anyway and that is when the failure happens

Reply:
It's strange because even `java.gradle` module contains wrapper instructions to download `gradle 7.5` .... do those tests pass ?

Reply:
Interesting. MN tests fail me even locally, while Gradle Java test *did download* gradle 7.5 distribution even though it has the same server in its wrapper.properties (`https://services.gradle.org/distributions/gradle-7.5-bin.zip vs. `https://services.gradle.org/distributions/gradle-7.6.2-bin.zip`).

Reply:
yeah its easily reproducible locally. All I had to do is to clear my `.gradle` folder (I kept the properties file).

Reply:
Yes, but any idea, why the Tooling API downloads one version, but is unable to download another version from *the same* server ?

Reply:
no. I also don't know why this all started a few days ago. I think the important little victory here is that the exception is logged, this should make issues like this less mysterious in future. This might also help to explain why the tests were so unreliable in past.

also, technically its not one server, its four: https://www.ssllabs.com/ssltest/analyze.html?d=services.gradle.org

Reply:
... neither of them supports TLS1.3

Reply:
someone pressed the wrong button in the admin console

Reply:
BTW I wireshark-captured the communication; what's funny is that the *client* (that is my laptop) reports **different** set of available cipher suites when connecting for gradle 7.5 download (from java.gradle test suite) than when connecting for gradle 7.6.2 (micronaut test suite).

Reply:
well the cert is from cloudflare, the rest of the infra probably too (I haven't checked). So all machines are going to be virtual, load balanced and ddos mitigated. So file A can be on the other side of the world than file B.

Do I interpret this as go-ahead with merge?

Reply:
Probably yes. I'd suggest to document the requirement to set that property in gradle's `arch.xml` or somewhere, since any testsuite that uses gradle project as test data may suffer from the same mystery.

Re. cipher suites: according to the capture, the client (that is: our CI machine)  **started** the handshake by offering different set of capabilities - there was no message whatsoever from the server before that (all payloads before Client Hello were zero length). The certificate is sent after server hello (which happens after client hello).

So the difference was IMHO not caused by potentially load-balanced server sending something.

Reply:
I am not planning to document this since this is a hack and has to be removed asap.